### PR TITLE
Add default values support for structs

### LINKED ISSUE #2209: Implement default value support for structs
Here is a sample struct.  You can only provide a default value to struct fields with the type value type.

struct Person {
    int a;
    string c = "hello";
    int k = 5;
}

### PR DESCRIPTION
**Pull request by SupunS**
Fix #2209

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java [hunk 1]
```diff
@@ -1064,28 +1066,39 @@ private void assignValueToVarRefExpr(BValue rValue, VariableRefExpr lExpr) {
...
+        return bStruct;
+    }
+
+    /**
+     * Invoke the init function of the struct. This will populate the default values for struct fields.
+     * 
+     * @param structDef Struct definition
+     * @param structMemBlock Memory block to be assigned for the new struct instance
+     */
+    private void invokeStructInitFunction(StructDef structDef, BValue[] structMemBlock) {
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java]: Please move this function to the private function section

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java [hunk 2]
```diff
@@ -1064,28 +1066,39 @@ private void assignValueToVarRefExpr(BValue rValue, VariableRefExpr lExpr) {
...
+        Expression[] argExprs = structInitExpr.getArgExprs();
         for (int i = 0; i < argExprs.length; i++) {
             MapStructInitKeyValueExpr expr = (MapStructInitKeyValueExpr) argExprs[i];
             VariableRefExpr varRefExpr = (VariableRefExpr) expr.getKeyExpr();
             StructVarLocation structVarLoc = (StructVarLocation) (varRefExpr).getVariableDef().getMemoryLocation();
             structMemBlock[structVarLoc.getStructMemAddrOffset()] = expr.getValueExpr().execute(this);
         }
 
-        return new BStruct(structDef, structMemBlock);
+        BStruct bStruct = new BStruct(structDef, structMemBlock);
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java]: These two lines are redundant.  You can just return new bstruct

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java [hunk 3]
```diff
@@ -1198,6 +1211,10 @@ private void setUnitValue(BValue rValue, BStruct lExprValue, int memoryLocation,
 
         // Get the arrays/map value from the mermory location
         BValue arrayMapValue = lExprValue.getValue(memoryLocation);
+        
+        if (arrayMapValue == null) {
+            throw new BallerinaException("field '" + fieldExpr.getSymbolName() + "' is not initialized");
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java]: IMHO, we should say the value is null.

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java (hunk 1 above)
Comment (SupunS) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/BLangExecutor.java]: done

**Pull request merged**